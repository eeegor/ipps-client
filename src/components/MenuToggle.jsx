import React from 'react';
import classnames from 'classnames';
import './MenuToggle.scss';

export const MenuToggle = props => {
  const { className, onClick, isOpen } = props;
  const classNames = classnames(
    'menu-toggle',
    className,
    isOpen && `menu-toggle--open`
  );
  return (
    <div className={classNames} onClick={event => onClick(event)}>
      {isOpen ? 'X' : 'Menu'}
    </div>
  );
};
