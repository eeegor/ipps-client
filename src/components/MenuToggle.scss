@import '../variables';

.menu-toggle {
  align-items: center;
  background: $color--primary;
  border-radius: 0.25rem;
  // box-shadow: 0 0.25rem 2rem 0 rgba(#333, 0.4);
  border: 2px solid $color--primary;
  color: #fff;
  display: flex;
  font-size: 0.8rem;
  font-weight: 700;
  justify-content: center;
  letter-spacing: 0.15rem;
  position: fixed;
  text-transform: uppercase;
  z-index: 100000;

  height: 2.5rem;
  width: 5rem;
  right: 0.75rem;
  top: 0.75rem;
  transition: all 0.2s ease-in-out;
  @media (min-width: $size--tablet) {
    height: 3rem;
    width: 6rem;
    right: 1rem;
    top: 1rem;
  }

  &--open {
    color: #fff;
    // color: $color--primary;
    background: rgba($color--primary, 0.5);
    border-color: #fff;
    box-shadow: 0 0.25rem 2rem 0 rgba($color--primary, 0.4);
  }
}
