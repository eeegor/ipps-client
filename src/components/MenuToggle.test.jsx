import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { MenuToggle } from './MenuToggle';

describe('MenuToggle', () => {
  it('renders markup correctly', () => {
    const wrapper = <MenuToggle>Hello</MenuToggle>;
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should create a isOpen className', () => {
    const wrapper = shallow(<MenuToggle isOpen={true}>Hello</MenuToggle>);
    expect(wrapper.prop('className')).toContain('menu-toggle--open');
  });

  it('click event works', () => {
    const callback = jest.fn();
    const wrapper = mount(<MenuToggle onClick={callback}>Hi</MenuToggle>);
    wrapper.simulate('click');
    expect(callback).toHaveBeenCalledTimes(1);
  });
});
