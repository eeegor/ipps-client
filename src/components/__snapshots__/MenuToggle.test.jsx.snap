// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuToggle renders markup correctly 1`] = `
<div
  className="menu-toggle"
  onClick={[Function]}
>
  Menu
</div>
`;
